ios/Midnight.swift

```diff
@@ -0,0 +1,47 @@
+// Midnight.swift
+
+import Foundation
+
+@objc(Midnight)
+class Midnight: RCTEventEmitter {
+    // MARK: RN Exposed Methods
+    
+    // This method should be removed when updating UI side as its more of a placeholder
+    @objc(multiply:withB:resolver:rejecter:)
+    func multiply(
+        _ a: Double,
+        withB b: Double,
+        resolver resolve: RCTPromiseResolveBlock,
+        rejecter reject: RCTPromiseRejectBlock
+    ) {
+        resolve(a * b)
+    }
+
+    @objc func subscribe() {
```

Comment:
I think we want `subscribe` and `unsubscribe` to instead be `startObserving` and `stopObserving`: https://github.com/facebook/react-native/blob/f7c209cd54cf336a8c500077537f5f10f8eda360/React/Modules/RCTEventEmitter.h#L37

I _think_ if you do it this way, then you don't have to manually call `subscribe` or `unsubscribe` on the javascript side, the event emitter handles that for you